sgkit_vcf/utils.py

```diff
@@ -23,9 +28,19 @@ def chunks(iterator: Iterator[T], n: int) -> Iterator[Iterator[T]]:
         yield itertools.chain([first], rest_of_chunk)  # concatenate the first item back
 
 
-def get_file_length(path: Path) -> int:
+def get_file_length(
+    path: PathType, storage_options: Optional[Dict[str, str]] = None
+) -> int:
     """Get the length of a file in bytes."""
-    return path.stat().st_size
+    if isinstance(path, Path):
```

Comment:
What's the advantage of checking the type over just using fsspec for all code paths? Is  fsspec substantially slower? Might be worth considering simplifying to a single code path if it's possible.